terraform.yml

```diff
@@ -0,0 +1,140 @@
+parameters:
+  - name: original_node_count
+    displayName: "Original Node Count"
+    type: number
+    default: 3
+    values:
+      - 3
+      - 14
+  - name: scale_up_node_count
+    displayName: "Scale Up Node Count +"
+    type: number
+    default: 2
+    values:
+      - 2
+  - name: region
+    displayName: "Region to run pipeline"
+    type: string
+    default: "eastus"
+    values:
+      - "eastus"
+      - "eastus2"
+      - "westus"
+      - "westus2"
+      - "westeurope"
+  - name: run_terraform
+    displayName: "Select terraform example to run"
+    type: string
+    default: "azureblobfiler"
+    values:
+      - "1-filer"
+      - "proxy"
+      - "azureblobfiler"
+  - name: image
+    displayName: "Select vfxt image to run. Defaults to latest."
+    type: string
+    default: "microsoft-avere:vfxt:avere-vfxt-node:6.0.0"
+    values:
+      - "microsoft-avere:vfxt:avere-vfxt-node:6.0.0"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.61"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.51"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.43"
+      - "microsoft-avere:vfxt:avere-vfxt-node:5.3.42"
+  - name: run_vdbench
+    displayName: "Run vdbench - only runs with azureblobfiler"
+    type: boolean
+    default: true
+  - name: vdbench_config
+    displayName: "vdbench config - only runs with azureblobfiler"
+    type: string
+    default: "inmem"
+    values:
+      - "ondisk"
+      - "inmem"
+  - name: run_scale_up
+    displayName: "Run scale-up"
+    type: boolean
+    default: true
+  - name: run_scale_down
+    displayName: "Run scale-down (only runs if scale-up is selected)"
+    type: boolean
+    default: false
+  - name: run_destory_and_cleanup
+    displayName: "Run terraform destroy and rg cleanup."
+    type: boolean
+    default: true
+  - name: pass_in_ssh_key
+    displayName: "Pass in public ssh key - make sure to add variable: PUBLIC_KEY"
+    type: boolean
+    default: false
+  
+jobs:
+- job: Terraform_vFXT
+  timeoutInMinutes: 180
+  cancelTimeoutInMinutes: 40
+  pool:
+    vmImage: ubuntu-latest
+  steps:  
+  - template: templates/setup_ssh.yml
+  - template: templates/setup_envars.yml
+    parameters:
+      run_terraform: ${{ parameters.run_terraform }}
+      region: ${{ parameters.region }}
+  - template: templates/terraform_setup.yml
+    parameters:
+      original_node_count: ${{ parameters.original_node_count }}
+      run_terraform: ${{ parameters.run_terraform }}
+      image: ${{ parameters.image }}
+  
+  - template: templates/terraform_cmd.yml
+    parameters: 
+      run_terraform: ${{ parameters.run_terraform }}
+      cmd_terraform: 'apply'
+      condition: "succeeded()"
```

Comment:
Minor/nitpick: quotes not needed. But they don't harm anything if you want to disregard.